onnxruntime/core/framework/execution_frame.cc

```diff
@@ -86,13 +86,54 @@ Status IExecutionFrame::GetOrCreateNodeOutputMLValue(int index, const TensorShap
                     " Requested shape:", shape ? shape->ToString() : "null");
       }
     } else {
+      VerifyOutputSizes(output_index, node, shape);
```

Comment:
I would rather do this check conditionally than paying the price *always*. An invalid ONNX model is probably not the norm and we shouldn't penalize the normal use case. Note that this verification is done for every single output index for every single node for every single kernel in the graph.

Reply:
Added a check for whether a node is an internal or final output node, before calling the verification method.